numpyro/distributions/transforms.py

```diff
@@ -583,6 +582,50 @@ def log_abs_det_jacobian(self, x, y, intermediates=None):
             )
 
 
+class L1BallTransform(Transform):
+    r"""
+    Transforms a uncontrained real vector :math:`x` into the unit L1 ball.
+    """
+    domain = constraints.real_vector
+    codomain = constraints.l1_ball
+
+    def __call__(self, x):
+        t = jnp.tanh(x)
+        # make sure that t in (-1, 1)
+        eps = jnp.finfo(t.dtype).eps
+        t = jnp.clip(t, a_min=(-1 + eps), a_max=(1 - eps))
+
+        # apply stick-breaking transform
+        remainder = jnp.cumprod(1 - jnp.abs(t[..., :-1]), axis=-1)
+        pad_width = [(0, 0)] * (t.ndim - 1) + [(1, 0)]
+        remainder = jnp.pad(remainder, pad_width, mode="constant", constant_values=1.0)
+        return t * remainder
+
+    def _inverse(self, y):
```

Comment:
If there is a $|t_i| = 1$, the inverse will become `NaN` for all positions after i. With clipping, it becomes `inf`. 

``` python
# make sure that t in (-1, 1)
t = jnp.array([-.8, .9, -1., -.5])  # jnp.tanh(-10.) will produce -1.

eps = jnp.finfo(t.dtype).eps
t = jnp.clip(t, a_min=(-1 + eps), a_max=(1 - eps))

# apply stick-breaking transform
remainder = jnp.cumprod(1 - jnp.abs(t[..., :-1]), axis=-1)

pad_width = [(0, 0)] * (t.ndim - 1) + [(1, 0)]
remainder = jnp.pad(remainder, pad_width, mode="constant", constant_values=1.0)
y = t * remainder
remainder = 1 - jnp.cumsum(jnp.abs(y[..., :-1]), axis=-1)

pad_width = [(0, 0)] * (y.ndim - 1) + [(1, 0)]
remainder = jnp.pad(remainder, pad_width, mode="constant", constant_values=1.0)
t = y / remainder

# inverse
print(t)  # [-0.8, 0.8999999, -1.0000011,  -inf]
```
If you clip the `x` param to `[-6,6]` in the forward transformation (`__call__`),`tanh(x).abs() <= 0.9999876`  so it will not  produce zeros in the `remainder`.  Something like
``` python
 def __call__(self, x):
        t = jnp.tanh(jnp.clip(x,a_min=-6., a_max=6.))
        ...
```

Let me know what you think.

Reply:
Thanks, I think that during accumulation `1 - jnp.cumsum(jnp.abs(y[..., :-1]), axis=-1)`, the remainder can be `0`. I will add a clip for it.